### DISCUSSION TITLE
Universal-Updater for DS(i) archives

### DISCUSSION THREAD

**Discussion by spellboundtriangle** [Universal-Team/Universal-Updater#131] [General]
I would like to preface that, I really don't know where to put this, as it more or less concerns Universal-Updater, but also the database, and various DS-Homebrew projects.
I originally brought this up in the DS(i) Mode Hacking Discord server, but I thought it would be best to bring it somewhere more long-term where it can be more easily discussed over time. It might be more appropriate as an issue, but I'm not sure. If you know where it might be better suited, please let me know.

Anyways, what I wanted to bring up is that from my understanding, extracting 7z archives is infeasible on DS(i) due to memory limitations, and this is one of the concerns for Universal-Updater for DS(i)'s development. However, zip should not be of much issue for it to handle.

Looking at Universal-DB, only a few (about seven) existing projects in the DS category are using 7z, the main ones being TWiLight Menu++, GodMode9i, and the DS Forwarder Pack. The rest are standalone files, or zips.

I would think the issue of 7z archives (for actively updated projects like TWiLight Menu++, specifically) would be easily worked around by using temporary zips of builds (both nightly and release) specifically for use with Universal-Updater on DS and DSi. These zips would be replaced upon becoming redundant, similar to how nightly builds are handled in the [extras repository](https://github.com/Universal-Team/extras).
<ins>Of course, this is something for TWiLight Menu++ developers to take care of</ins>, unless you want to take on rehosting those ~100-300 MB of zips automatically
The benefits of using temporary zips would allow the following:
- 7z archives would not be a concern for the development of Universal-Updater for DS(i)
- The better compressed 7z archives for manual and 3DS installations may continue to be used
- Because the zips will only be up for as long as they need to be, and Universal-Updater can't download old releases by default anyways, it shouldn't have any large impact on storage on GitHub, while still preserving the originals in their 7z forms

If you have any thoughts on this subject, please share them. I kind of rushed putting this together, hopefully it's coherent enough.

**Comment by Epicpkmn11** [comment]
7z isn't really the biggest thing holding it up, iirc zip is already working in the latest build or if not I got it working somewhere. It's more just dsiwifi being iffy make it a pain to work on and the general bugginess.

If StackZ wants to work on the 3DS part and ignore the DSi part since that's not working for him I'd be happy to port the 3DS side change to DSi, but he hasn't been motivated to work on UU. I may pick it back up myself, but I was never really the main dev...

It honestly might be better to more or less scrap the current rewrite and just start on a completely separate DSi version, it seemed like a good idea at the time to share as much code as possible but it's kinda just made both versions harder to develop I think...

For the main topic of this, yeah I think it'd work to just change most things to uncompressed or just a zip, and could always upload both for compatibility – nds-bootstrap already does.
